src/codeu/chat/util/Serializers.java

```diff
@@ -150,6 +151,31 @@ public void write(OutputStream out, Collection<T> value) throws IOException {
       }
     };
   }
+  
+  public static <K, V> Serializer<HashMap<K, V>> hashmap(final Serializer<K> keySerializer, final Serializer<V> valueSerializer) {
```

Comment:
Done. Can you make any suggestions on how I would write tests for this? I don't really undersatand how to simulate the server environment since the streams come from the server wherever this serializer is used. Thanks.